src/Extensions/SwiftypeSiteTreeCrawlerExtension.php

```diff
@@ -25,14 +26,18 @@ public function onAfterPublish(&$original): void
     {
         parent::onAfterPublish($original);
 
-        $this->forceSwiftypeIndex();
+        $this->withVersionContext(function() {
+            $this->forceSwiftypeIndex();
+        });
     }
 
     public function onAfterUnpublish(): void
     {
         parent::onAfterUnpublish();
 
-        $this->forceSwiftypeIndex();
+        $this->withVersionContext(function() {
+            $this->forceSwiftypeIndex();
+        });
```

Comment:
Forcing the stage to `LIVE` `onAfterUnpublish` is going to result in your simply getting the base URL, not the URL of the page you want to purge. Perhaps this hook should actually be on `onBeforeUnpublish`, or, we want to get `stage=Stage`, but drop the params?